### PR TITLE
Analyzer updated and SingleLepton dataset added

### PR DESCRIPTION
**Pull request by clacaputo**
### modified:   test/zhtautau/samples.json
- Added single lepton (ele/mu) 2016 datasets
- DYJets NLO restored

### new file:   test/zhtautau/checkMissingFiles.py
- python script used for checking number of missing root file after submit.sh

### modified:   bin/zhtautau/runZHTauTauAnalysis.cc
- add trigger paths for single lepton, fixed filter for data
- add WZ EWK corrections
- BTag
   * WP updated to Moriond17 prescriptions
   * Systematic treatment updated
- MET Filter for 2016 Data introduced
- Leptons correction
   * Corrections applied to LooseLeptons
   * Reintroduced electron object duplication after corrections
- Kinematic Cuts on leptons updated (mu: 25GeV, ele: 30GeV)
- ZptCut request before selecting 3rd and 4th lepton
- ID-ISO Scale factor for higgs leptons computed for LOOSE WP

### CODE REVIEW

File: bin/zhtautau/runZHTauTauAnalysis.cc [hunk 1]
```diff
@@ -1308,7 +1428,7 @@ int main(int argc, char* argv[])
 
 
         //SIGNAL ANALYSIS Z+2Leptons  (no systematics taken into account here)
-        if(passZmass && (int)selLeptons.size()>=4){  //Request at least 4 leptons
+        if(passZmass && passZpt && (int)selLeptons.size()>=4){  //Request at least 4 leptons
```
Comment (mmusich) [inline bin/zhtautau/runZHTauTauAnalysis.cc:1431]: @clacaputo, why requiring the cut on the boson momentum at this stage?

File: bin/zhtautau/runZHTauTauAnalysis.cc [hunk 2]
```diff
@@ -1333,9 +1453,9 @@ int main(int argc, char* argv[])
             if(abs(selLeptons[dilLep1].pdgId())==11){HiggsShortId += 0;}else{HiggsShortId += 6;}
             switch(abs(higgsCandId)){
               case 11*11:  ChannelName  = "elel";  HiggsShortId+= 0; break;
-              case 11*13:  ChannelName  = "elmu";  HiggsShortId+= 1; break;
-              case 11*15:  ChannelName  = "elha";  HiggsShortId+= 2; break;
-              case 13*13:  ChannelName  = "mumu";  HiggsShortId+= 3; break;
+              case 13*13:  ChannelName  = "mumu";  HiggsShortId+= 1; break;
+              case 11*13:  ChannelName  = "elmu";  HiggsShortId+= 2; break;
+              case 11*15:  ChannelName  = "elha";  HiggsShortId+= 3; break;
```
Comment (mmusich) [inline bin/zhtautau/runZHTauTauAnalysis.cc:1458]: any rationale about the reordering of the final states?

**Review by mmusich — COMMENTED** [review]
Hi @clacaputo, thanks for the improvements, I've put two questions for my own understanding. Once answered I will merge.

File: bin/zhtautau/runZHTauTauAnalysis.cc (hunk 2 above)
Comment (clacaputo) [reply bin/zhtautau/runZHTauTauAnalysis.cc:1458]: For matching the bin names in the Yield histograms

File: bin/zhtautau/runZHTauTauAnalysis.cc (hunk 1 above)
Comment (clacaputo) [reply bin/zhtautau/runZHTauTauAnalysis.cc:1431]: The ptZ is not well modelled in NLO MC samples. The cut has been introduced to check whether it improves the Data/MC agreement

File: bin/zhtautau/runZHTauTauAnalysis.cc (hunk 2 above)
Comment (mmusich) [reply bin/zhtautau/runZHTauTauAnalysis.cc:1458]: checked that the new logic is correct. 
Thanks for fixing 

```
HiggsShortId:0  ->"OS eeee" 
HiggsShortId:1  ->"OS ee#mu#mu"          (was "OS eee#mu")
HiggsShortId:2  ->"OS eee#mu"            (was "OS eee#tau")
HiggsShortId:3  ->"OS eee#tau"           (was "OS ee#mu#mu")
HiggsShortId:4  ->"OS ee#mu#tau" 
HiggsShortId:5  ->"OS ee#tau#tau" 
HiggsShortId:6  ->"OS #mu#muee" 
HiggsShortId:7  ->"OS #mu#mu#mu#mu"      (was "OS #mu#mue#mu")  
HiggsShortId:8  ->"OS #mu#mue#mu" 	 (was "OS #mu#mue#tau") 
HiggsShortId:9  ->"OS #mu#mue#tau" 	 (was "OS #mu#mu#mu#mu")
HiggsShortId:10 ->"OS #mu#mu#mu#tau" 
HiggsShortId:11 ->"OS #mu#mu#tau#tau" 
HiggsShortId:12 ->"SS eeee" 
HiggsShortId:13 ->"SS ee#mu#mu"          (was "SS eee#mu")	 
HiggsShortId:14 ->"SS eee#mu" 		 (was "SS eee#tau")	
HiggsShortId:15 ->"SS eee#tau" 		 (was "SS ee#mu#mu")	
HiggsShortId:16 ->"SS ee#mu#tau" 	                        
HiggsShortId:17 ->"SS ee#tau#tau" 	                        
HiggsShortId:18 ->"SS #mu#muee" 	                        
HiggsShortId:19 ->"SS #mu#mu#mu#mu" 	 (was "SS #mu#mue#mu")  
HiggsShortId:20 ->"SS #mu#mue#mu" 	 (was "SS #mu#mue#tau") 
HiggsShortId:21 ->"SS #mu#mue#tau" 	 (was "SS #mu#mu#mu#mu")
HiggsShortId:22 ->"SS #mu#mu#mu#tau" 
HiggsShortId:23 ->"SS #mu#mu#tau#tau" 
```

**Pull request merged**